### DISCUSSION TITLE
HmIP BDT - Restore last dimmer level

### DISCUSSION THREAD

**Discussion by gerritb** [SukramJ/aiohomematic#736] [Ideas]
DONT'T DELETE THIS.

**Is your feature request related to a problem? Please describe.**
Homematic BDT can be set to restore the last dimmer level on the button turn on in CCU settings. The dimmer level is set to 100 when turning on in HA. Other light integrations like Yeelight and Hue also set the last dimmer level in HA. 

**Describe the solution you'd like**
Don't know if it's possible with the structure of the py classes in this project but it would be nice to have the last dimmer state restored when turning the lights on through HA. 

**Describe alternatives you've considered**

**Additional context**

**Comment by SukramJ** [comment]
> Homematic BDT can be set to restore the last dimmer level on the button turn on in CCU settings.

Please post a screenshot of this setting.

**Comment by gerritb** [comment]
Here you go. This is from the button up configuration in direct device connections that is automatically created after pairing the BDT with the CCU. 

![image](https://user-images.githubusercontent.com/440007/206658114-8cd10173-1613-4bfe-ae5c-4a04daa7866c.png)

**Reply by SukramJ** [reply]
Hue and Yeelight store the last value in the device, and HM uses a direct connection to store the last value.
The problem is that HM does not store the last value in the device, so we cannot use it.
We do not not store the any additional state in our own python classes except from the device it self.

If we did, then that would only ever be the last known status in HA. In addition, the device would differ in operation from the WebUI, and I would actually like to avoid that.

**Reply by gerritb** [reply]
Thanks, thought about the reason being something like that. Since HM can recover it, could it be somehow retrieved from the API? Not familiar with the integration and HM API but I guess I will start digging deeper.
